CMakeLists.txt

```diff
@@ -125,7 +150,12 @@ if(MSVC)
     endif()
 endif()
 
-find_package(CUDA)
+if (CONDA_ROOT)
+    find_package(CUDAToolkit)
```

Comment:
I think this should be unconditional for a recent CMake. 3.18, IIRC, is the lowest version that got this. 3.19 even supports CUDA as a project language. And then you do not need to `find_package(CUDA)`. As is, the `if` is kinda misleading: what is put inside it isn't because Conda, it's because CMake is recent.

Reply:
> 3.19 even supports CUDA as a project language.

It's CMake 3.8, I think. We are already using CUDA as a project language in k2.
 See https://github.com/k2-fsa/k2/blob/master/CMakeLists.txt#L46

I agree we should throw away `find_package(CUDAToolkit)` as well as `find_package(CUDA)`.

Reply:
Yeah I agree that'd be better.  I did try at one point to use CMake's cuda language support, but it was throwing some errors elsewhere, so I went with the more hacky approach just to get it working.  Would it be alright to switch over to using CUDA as a project language after this gets merged in, just in case those errors I hit haven't been fixed already by the work that @h-vetinari has done.

Reply:
@mmcauliffe, yes, I also do not think CUDA as a language should go into this changeset, there are too many little changes around enabling it, which won't affect the Conda-enabling things, so it's clearly separate. 

And, @csukuangfj, you cannot throw away `find_package(CUDAToolkit)` even if CUDA language is enabled. You still need the libraries. `find_package(CUDA)` is obsoleted by `find_package(CUDAToolkit)` though.